modules/capability/capability_test.go

```diff
@@ -3,101 +3,119 @@ package capability_test
+	cms := store.NewCommitMultiStore(db, log.NewNopLogger(), metrics.NewNoOpMetrics())
+	cms.MountStoreWithDB(suite.storeKey, storetypes.StoreTypeIAVL, db)
+	cms.MountStoreWithDB(suite.memStoreKey, storetypes.StoreTypeMemory, db)
+	cms.MountStoreWithDB(suite.mockMemStoreKey, storetypes.StoreTypeMemory, db)
+
+	err := cms.LoadLatestVersion()
+	suite.Require().NoError(err)
+
+	return sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger())
 }
 
 // The following test case mocks a specific bug discovered in https://github.com/cosmos/cosmos-sdk/issues/9800
 // and ensures that the current code successfully fixes the issue.
+// This test emulates statesync by firstly populating persisted state by creating a new scoped keeper and capability.
+// In-memory storage is then discarded by creating a new capability keeper and app module using a mock memstore key.
+// BeginBlock is then called to populate the new in-memory store using the persisted state.
 func (suite *CapabilityTestSuite) TestInitializeMemStore() {
-	sk1 := suite.keeper.ScopeToModule(banktypes.ModuleName)
+	// create a scoped keeper and instantiate a new capability to populate state
+	scopedKeeper := suite.keeper.ScopeToModule(banktypes.ModuleName)
 
-	cap1, err := sk1.NewCapability(suite.ctx, "transfer")
+	cap1, err := scopedKeeper.NewCapability(suite.ctx, "transfer")
 	suite.Require().NoError(err)
 	suite.Require().NotNil(cap1)
 
-	// mock statesync by creating new keeper that shares persistent state but loses in-memory map
-	newKeeper := keeper.NewKeeper(suite.cdc, suite.app.GetKey(types.StoreKey), suite.app.GetMemKey(memStoreKey))
-	newSk1 := newKeeper.ScopeToModule(banktypes.ModuleName)
+	// mock statesync by creating a new keeper and module that shares persisted state
+	// but discards in-memory map by using a mock memstore key
+	newKeeper := keeper.NewKeeper(suite.cdc, suite.storeKey, suite.mockMemStoreKey)
+	newModule := capability.NewAppModule(suite.cdc, *newKeeper, true)
+
+	// reassign the scoped keeper, this will inherit the the mock memstore key used above
+	scopedKeeper = newKeeper.ScopeToModule(banktypes.ModuleName)
 
-	// Mock App startup
-	ctx := suite.app.BaseApp.NewUncachedContext(false, tmproto.Header{})
+	// seal the new keeper and ensure the in-memory store is not initialized
 	newKeeper.Seal()
-	suite.Require().False(newKeeper.IsInitialized(ctx), "memstore initialized flag set before BeginBlock")
+	suite.Require().False(newKeeper.IsInitialized(suite.ctx), "memstore initialized flag set before BeginBlock")
 
-	// Mock app beginblock and ensure that no gas has been consumed and memstore is initialized
-	ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{}).WithBlockGasMeter(sdk.NewGasMeter(50))
+	// add a new block gas meter to the context
+	ctx := suite.ctx.WithBlockGasMeter(storetypes.NewGasMeter(50))
 
+	prevGas := ctx.GasMeter().GasConsumed()
 	prevBlockGas := ctx.BlockGasMeter().GasConsumed()
-	prevGas := ctx.BlockGasMeter().GasConsumed()
```

Comment:
`prevGas` here was incorrect, it was being set to the same value as `prevBlockGas`